### PR TITLE
chore: fix various build/compilation warnings

### LINKED ISSUE #723: Surefire 3.0.0-M4 passing even if a build step throws an exception
The XsltProcessor is throwing NoSuchMethodError when the app is augmented for testing (the same exception is mentioned in #721 for the dev mode). However with maven-surefire-plugin 3.0.0-M4, this exception is swallowed and the whole build is passing and 0 tests run are reported:

```
[INFO] Running org.apache.camel.quarkus.component.xslt.it.XsltTest
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.863 s - in org.apache.camel.quarkus.component.xslt.it.XsltTest
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
```

When the maven-surefire-plugin is downgraded to 2.22.1, it fails as expected:

```
[INFO] Running org.apache.camel.quarkus.component.xslt.it.XsltTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.871 s <<< FAILURE! - in org.apache.camel.quarkus.component.xslt.it.XsltTest
[ERROR] org.apache.camel.quarkus.component.xslt.it.XsltTest  Time elapsed: 1.871 s  <<< ERROR!
java.lang.RuntimeException: 
java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step org.apache.camel.quarkus.component.xslt.deployment.XsltProcessor#xsltResources threw an exception: java.lang.NoSuchMethodError: javax.xml.transform.TransformerFactory.newInstance(Ljava/lang/String;Ljava/lang/ClassLoader;)Ljavax/xml/transform/TransformerFactory;
        at org.apache.camel.quarkus.support.xalan.XalanTransformerFactory.<init>(XalanTransformerFactory.java:41)
...
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   XsltTest » Runtime java.lang.RuntimeException: io.quarkus.builder.BuildExcepti...
[INFO] 
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
```

I have just reported this in Quarkus: https://github.com/quarkusio/quarkus/issues/7203

### PR DESCRIPTION
**Pull request by lburgazzoli**
Fixes #723

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -192,12 +192,16 @@
...
         <maven-resources-plugin.version>3.3.1</maven-resources-plugin.version>
         <maven-scm-plugin.version>2.0.0</maven-scm-plugin.version>
-        <maven-shade-plugin.version>3.4.1</maven-shade-plugin.version>
+        <maven-shade-plugin.version>3.5.0</maven-shade-plugin.version>
         <!-- NOTE: We pin to this version due to https://github.com/apache/camel-quarkus/issues/723 -->
         <maven-surefire-plugin.version>2.22.2</maven-surefire-plugin.version>
         <os-maven-plugin.version>${os-maven-plugin-version}</os-maven-plugin.version>
         <protobuf-maven-plugin.version>${protobuf-maven-plugin-version}</protobuf-maven-plugin.version>
         <rpkgtests-maven-plugin.version>1.0.0</rpkgtests-maven-plugin.version>
+        <maven-clean-plugin.version>3.2.0</maven-clean-plugin.version>
```
Comment (jamesnetherton) [inline pom.xml:201]: Sorry to nitpick - do you mind moving these maven-*-plugin props up with the others.

Comment (lburgazzoli) [reply pom.xml]: done

**Review by oscerd — APPROVED** [review]

**Comment by lburgazzoli** [comment]
@essobedo I'm facing some `ClassNotFoundException` related issues here about Groovy and jOOR when running the Native tests, like:

```
[error]: Build step org.apache.camel.quarkus.component.groovy.deployment.GroovyProcessor#configureLanguage threw an exception: java.lang.IllegalStateException: java.lang.ClassNotFoundException: org.apache.camel.quarkus.component.groovy.generated.Expression668axhkmpOfBwe2iixBEtT194721N7sqMJrOlgVZAJVxs61
```

```
[error]: Build step org.apache.camel.quarkus.component.joor.deployment.JoorProcessor#configureLanguage threw an exception: java.lang.IllegalStateException: java.lang.ClassNotFoundException: org.apache.camel.quarkus.component.joor.generated.Expression78EQYvzpgOCH7d9dbRcJQ89Oc7j47VuG0Nph92qsYLags61
```

I don't yet know how things works under the hood so any help would be appreciated :)

**Comment by essobedo** [comment]
> @essobedo I'm facing some `ClassNotFoundException` related issues here about Groovy and jOOR when running the Native tests, like:
> 
> ```
> [error]: Build step org.apache.camel.quarkus.component.groovy.deployment.GroovyProcessor#configureLanguage
> threw an exception: java.lang.IllegalStateException: java.lang.ClassNotFoundException:
> org.apache.camel.quarkus.component.groovy.generated.Expression668axhkmpOfBwe2iixBEtT194721N7sqMJrOlgVZAJVxs61
> ```
> 
> ```
> [error]: Build step org.apache.camel.quarkus.component.joor.deployment.JoorProcessor#configureLanguage 
> threw an exception: java.lang.IllegalStateException: java.lang.ClassNotFoundException:
> org.apache.camel.quarkus.component.joor.generated.Expression78EQYvzpgOCH7d9dbRcJQ89Oc7j47VuG0Nph92qsYLags61
> ```
> 
> I don't yet know how things works under the hood so any help would be appreciated :)

It is because you replaced `recorderContext.classProxy` with `Class.forName` in the Groovy and Joor processors, I already tried it before as it is supposed to replace it but it did not work as you can see. Those classes are generated classes based on expressions extracted from the route definitions

**Comment by lburgazzoli** [comment]
> ecorderContext.classProxy

ok so I will add a comment but I guess we whould report this to quarkus ? otherwhise the method may disappear at some point I guess

**Comment by essobedo** [comment]
> > ecorderContext.classProxy
> 
> ok so I will add a comment but I guess we whould report this to quarkus ? otherwhise the method may disappear at some point I guess

Yes, I guess so too, I can add it to my to-do list. I don't know if it is a Quarkus issue or a misusage, I could not investigate more so far.

File: extensions/management/runtime/src/main/resources/META-INF/quarkus-extension.yaml [hunk 2]
```diff
@@ -25,9 +25,8 @@ name: "Camel Management"
 description: "JMX management strategy and associated managed resources"
 icon-url: "https://camel.apache.org/_/img/logo-d-f21b25ba38.svg"
 metadata:
-  unlisted: true
```
Comment (ppalaga) [inline extensions/management/runtime/src/main/resources/META-INF/quarkus-extension.yaml:28]: Is JMX supported on GraalVM now? I know it was planned, I just want to make sure it happened already.

Comment (essobedo) [reply extensions/management/runtime/src/main/resources/META-INF/quarkus-extension.yaml:28]: Yes it is supported starting from v 23 but it is experimental

Comment (essobedo) [reply extensions/management/runtime/src/main/resources/META-INF/quarkus-extension.yaml:28]: It has been added by this PR https://github.com/oracle/graal/issues/4732

File: integration-test-groups/aws2/aws2-s3/src/main/java/org/apache/camel/quarkus/component/aws2/s3/it/Aws2S3Resource.java [hunk 3]
```diff
@@ -143,11 +141,15 @@ public Response read(@PathParam("bucketName") String bucketName, @PathParam("key
...
+        final List<?> objects = producerTemplate.requestBody(
                 componentUri(AWS2S3Operations.listObjects),
                 null,
                 List.class);
-        return objects.stream().map(S3Object::key).collect(Collectors.toList());
+
+        return objects.stream()
+                .map(S3Object.class::cast)
+                .map(S3Object::key)
+                .collect(Collectors.toList());
```
Comment (ppalaga) [inline integration-test-groups/aws2/aws2-s3/src/main/java/org/apache/camel/quarkus/component/aws2/s3/it/Aws2S3Resource.java:152]: 👌

**Comment by lburgazzoli** [comment]
> I don't know if it is a Quarkus issue or a misusage

Openend an issue to track this work: https://github.com/apache/camel-quarkus/issues/5056

**Review by ppalaga — APPROVED** [review]

**Comment by lburgazzoli** [comment]
@ppalaga @jamesnetherton let me know if there is anything you want me to change before merging it

**Review by jamesnetherton — APPROVED** [review]
LGTM

**Review by zbendhiba — APPROVED** [review]

**Pull request merged**